osutil/disks/luks.go

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var (
+	// this regexp is for the DM_UUID udev property, or equivalently the dm/uuid
+	// sysfs entry for a luks2 device mapper volume dynamically created by
+	// systemd-cryptsetup when unlocking
+	// the actual value that is returned also has "-some-name" appended to this
+	// pattern, but we delete that from the string before matching with this
+	// regexp to prevent issues like a mapper volume name that has CRYPT-LUKS2-
+	// in the name and thus we might accidentally match it
+	// see also the comments in DiskFromMountPoint about this value
+	luksUUIDPatternRe = regexp.MustCompile(`^CRYPT-LUKS2-([0-9a-f]{32})$`)
+)
+
+// TODO: should this specific handler live in a different package? Probably
+// fine for now to keep living in disks...
```

Comment:
this is fine, we can drop the TODO

Reply:
removed in https://github.com/snapcore/snapd/pull/11035